### DISCUSSION TITLE
Have issues when uploading files that replace existing file in storage

### DISCUSSION THREAD

**Discussion by EfosaSO** [supabase/supabase#2679] [Questions]
I've created that policy but still get the row-level security issue and I still can't see the policy on my dashboard.

This is my error message: "new row violates row-level security policy (USING expression) for table \"objects\""

**Comment by dshukertjr** [comment]
Hi @EfosaSO !

It sounds like you need to set a row level security policy on Update on `objects` table. You can set it from policy editor in your Supabase console!
<img width="1373" alt="Screen Shot 2021-08-02 at 20 40 44" src="https://user-images.githubusercontent.com/18113850/127856958-a404aa1e-aa02-4531-950f-e6d43df4d138.png">

Let me know if this works!

**Reply by EfosaSO** [reply]
So it was the `ALL` access I was missing, I think the docs need to be updated [here ](https://supabase.io/docs/reference/javascript/storage-from-upload#notes) to reflect that. Because it only mentions `INSERT`

**Reply by inian** [reply]
Are you using the upsert flag here @EfosaSO?

**Reply by EfosaSO** [reply]
Yes, I was using the upsert flag

**Reply by inian** [reply]
Hmm, I think for upsert both insert and select policy is required. I will update the docs.

**Reply by EfosaSO** [reply]
Just tried insert & select, ran into the issue from before

**Reply by inian** [reply]
ah sorry...can you try insert and update permissions instead?

**Reply by EfosaSO** [reply]
I tried that too and it didn't work, only ALL

**Reply by wadclapp** [reply]
For update, when using upsert, did you include a `USING` expression?

**Reply by SleeplessFox** [reply]
What does it mean. "did you include a USING expression'? How? I have the same problem.
